dgl_sparse/src/matmul.cc

```diff
@@ -98,6 +98,53 @@ torch::Tensor SDDMMNoAutoGrad(
   return ret;
 }
 
+torch::Tensor BroadcastOpNoAutoGrad(
+    const c10::intrusive_ptr<SparseMatrix>& sparse_mat, torch::Tensor e,
+    torch::Tensor v, const std::string& op) {
+  const int64_t out_row = sparse_mat->nnz();
+  const std::vector<int64_t> shape({out_row, e.size(1)});
+  auto ret = torch::zeros(shape, e.options());
+
+  auto dgl_e = TorchTensorToDGLArray(e);
+  auto dgl_v = TorchTensorToDGLArray(v);
+  auto dgl_ret = TorchTensorToDGLArray(ret);
+
+  // The format for calculation will be chosen in the following order: CSR,
+  // COO. CSR is created if the sparse matrix only has CSC format.
```

Comment:
The format selection for SDDMM should prioritize COO. The change should be easy: check if COO exists first.

Reply:
updated